File: day2.py

if __name__ == '__main__':
    with open("./inputs/day2.txt", 'r') as file:
        data = [line.strip().split(" ") for line in file.readlines()]
        data = [(x, int(y)) for [x, y] in data]
    print(data)
    horizontal = 0
    vertical = 0
    aim = 0
    depth = 0

    for move, measure in data:
        if move == 'forward':
            horizontal += measure
            depth += measure*aim
        elif move == 'up':
            vertical -= measure
            aim -= measure
        elif move == 'down':
            vertical += measure
            aim += measure
    print("Depth (part1)",horizontal*vertical, "Depth (part2): ", depth*horizontal)

File: 2021/day2.cpp
#include <cmath>
#include <fstream>
#include <iostream>
#include <vector>
#include <algorithm>

/* Advent of Code 2021 Day 02  */

int main(int argc, char * argv[]) {
    std::string input = "./inputs/day2.txt";
    std::fstream file(input);

    int horizontal, vertical, aim, depth = 0;
    std::string command;
    int measure;
    while(file >> command >> measure){
        if(command.compare("forward") == 0){
            horizontal += measure;
            depth += measure*aim;
        }else if(command.compare("up") == 0) {
            vertical -= measure;
            aim -= measure;
        } else {
            vertical += measure;
            aim += measure;
        }
    }

    // Part 1
    std::cout << "Part 1: " << horizontal*vertical << "\n";
    // Part 2
    std::cout << "Part 2: " << depth*horizontal << "\n";
}



